tiledb/api/c_api/filesystem/CMakeLists.txt

```diff
@@ -0,0 +1,47 @@
+#
+# tiledb/api/c_api/filesystem/CMakeLists.txt
+#
+# The MIT License
+#
+# Copyright (c) 2022 TileDB, Inc.
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+include(common NO_POLICY_SCOPE)
+
+list(APPEND SOURCES
+	filesystem_api.cc
+)
+gather_sources(${SOURCES})
+
+#
+# Object library for other units to depend upon
+#
+add_library(capi_filesystem_stub OBJECT ${SOURCES})
```

Comment:
Removed _stub from all libraries that compile without the dependency. The group and buffer libraries require it to avoid an error locating `spdlog.h`. I'll try replacing with the `baseline` dependency tomorrow to see if that clears things up to avoid the naming convention weirdness.

Reply:
Using baseline fixed the spdlog.h issue.